library/general/src/lib/ui/installation/layout.rb

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "ui/installation/layout_config"
+
+module UI
+  module Installation
+    # Class to define the wizard layout for the installation
+    #
+    # There are three possible layouts:
+    #
+    # * With a left sidebar to place the installation steps.
+    # * Without a sidebar but with the title of the dialogs on the left.
+    # * Without a sidebar but with the title of the dialogs on top.
+    #
+    # Moreover, for each layout, a top banner can be added.
+    #
+    # @example
+    #   layout1 = UI::Installation::Layout.with_steps
+    #   layout2 = UI::Installation::Layout.with_title_on_left
+    #   layout3 = UI::Installation::Layout.with_title_on_top
+    #
+    #   layout3.show_banner
+    #
+    #   layout1.open_wizard do
```

Comment:
To be honest I hope that this is not the main point to this layout. Why not add method also to Yast::Wizard? like OpenWithLayout? which just call this? or course lazy loading. As wizard is first class which you inspect if you want to open it.

Reply:
Well, the main points of this layout class are:

* Allow all possible configurations for the installation/firstboot in a object oriented way.
* Automatically configure the layout according to the control file.
* Abstract to the user about some low level details like which Wizard to open depending on the configuration.
* Abstract to the user about low level details like how to show/hide a banner.

Yes, maybe we can add *Wizard#OpenWithLayout(layout)*.